### ISSUE TITLE
Auto-correct cops for invalid syntax

### ISSUE THREAD

**Issue by joeldrapper** [rubocop/rubocop#14141]
## Is your feature request related to a problem? Please describe.

I’ve been trying to make a custom cop for [Phlex](https://www.phlex.fun) that can detect when you have ERB directly inside a Ruby method and convert it to Ruby.

**Example input:**
```
def view_template
  <h1>Hello <%= @name %></h1>
end
```

**Corrected:**
```ruby
def view_template
  h1 { "Hello #{@name}" }
end
```

However, I believe I’ve run into a limitation that Rubocop will never call custom cops when the Ruby syntax is already known to be invalid.

I’m pretty sure I can use Prism to extract the HTML here, even though it’s completely broken Ruby syntax. I just can’t find the right hook.

## Describe the solution you'd like

Would it be possible to have a new kind of cop that can be called upon to auto-correct Ruby files that have a syntax error?

## Describe alternatives you've considered

My only alternative I guess would be to write a separate program that wouldn't be triggered by Rubocop’s auto-correct process.

**Comment by Earlopain** [comment]
How about `on_other_file`? It gets called for files with invalid syntax and is also how `Lint/Syntax` is invoked. Although, I have no idea if/how autocorrect works inside of it.

I know that https://github.com/ksss/rubocop-on-rbs uses it to process rbs, you can probably check that project out to learn from.
